app/src/examples/ShareablesExample.tsx

```diff
@@ -0,0 +1,96 @@
+import { Button, StyleSheet, View } from 'react-native';
+
+import React from 'react';
+import { runOnJS, runOnUI } from 'react-native-reanimated';
+
+export default function ShareablesExample() {
+  return (
+    <View style={styles.container}>
+      <CyclicObjectDemo />
+      <InaccessibleObjectDemo />
+      <ArrayBufferDemo />
+      <TypedArrayDemo />
+      <DataViewDemo />
+    </View>
+  );
+}
+
+function CyclicObjectDemo() {
+  const handlePress = () => {
+    type RecursiveArray = (number | RecursiveArray)[];
+    const x: RecursiveArray = [];
+    x.push(1);
+    x.push(x);
+    runOnUI(() => {
+      console.log(x);
+    })();
+  };
+
+  return <Button title="Cyclic object" onPress={handlePress} />;
+}
+
+function InaccessibleObjectDemo() {
+  const handlePress = () => {
+    const x = new Set();
+    runOnUI(() => {
+      console.log(x);
+    })();
+  };
+
+  return <Button title="Inaccessible object" onPress={handlePress} />;
+}
+
+function ArrayBufferDemo() {
+  const handlePress = () => {
+    const ab = new ArrayBuffer(8);
+    const ta = new Uint8Array(ab);
+    ta[7] = 42;
+    function after() {
+      console.log(ta[7] === 42);
+    }
+    runOnUI(() => {
+      console.log(ab instanceof ArrayBuffer);
+      const ta = new Uint8Array(ab);
+      console.log(ta[7] === 42);
+      ta[7] = 123;
+      runOnJS(after)();
+    })();
+  };
+
+  return <Button title="ArrayBuffer" onPress={handlePress} />;
+}
+
+function TypedArrayDemo() {
+  const handlePress = () => {
+    const ta = new Uint32Array(8);
+    ta[7] = 1234567;
+    runOnUI(() => {
+      console.log(ta instanceof Uint32Array);
+      console.log(ta[7] === 1234567);
+    })();
+  };
+
+  return <Button title="TypedArray" onPress={handlePress} />;
+}
+
+function DataViewDemo() {
+  const handlePress = () => {
+    const buffer = new ArrayBuffer(16);
+    const dv = new DataView(buffer);
+    dv.setInt16(7, 12345);
+    runOnUI(() => {
+      console.log(dv instanceof DataView);
+      console.log(dv.getInt16(7) === 12345);
+    })();
+  };
+
+  return <Button title="DataView" onPress={handlePress} />;
+}
```

Comment:
Could you add another example that uses all data types listed in  VALID_ARRAY_VIEWS_NAMES? It could be simpletes. For example create a specific buffer and use it in the worklet.

Reply:
Done in ebc26986c454e7a4b4637f9a08d54905d7193862

Reply:
Do you have something that opens the example pages to see if they crash or not as a test? (asking because I'd be interested to setup something like this maybe in RN Skia (we do have e2e tests separately)).

Reply:
Not yet, but recently we've been talking with @piaskowyk about adding it. In our case, we could also overwrite the implementation `_getAnimationTimestamp()` function and test the animations in a frame-by-frame manner. We couldn't find any simple way for testing whole features since they are implemented in multiple programming languages at once and also heavily depend on the RN itself, so we would need some kind of E2E framework to run the feature tests. I've experimeneted with E2E tests for Reanimated in the past (see https://github.com/tomekzaw/reanimated-e2e/tree/appium) but they were too flaky and time-consuming so we didn't add them to our CIs.

Reply:
Not yet, but we're working on something like that. But now we do it manually before every release 😅

Reply:
On the Skia side, we found e2e frameworks to be also too "flaky" (more specifically we found that they were many too many assumptions and giving us enough control over the setup) and we have built our own e2e solution using the WebSocket API. This has worked really well for us.

1. People write a function in Jest: https://github.com/Shopify/react-native-skia/blob/main/package/src/renderer/__tests__/e2e/Point.spec.tsx#L8 and behind the scene we sent the source code of the function over the write to the RN app (the function needs to be pure).
2. On our example app, we have a screen that just waits and see if it receives tests to be executed on its side: https://github.com/Shopify/react-native-skia/blob/main/example/src/Tests/useClient.ts
3. Write now we use it to eval JS code on the RN side or doing a Skia drawing and returns an image. But we are thinking of extending it with more complex commands.